cdhist/cdhist.py

```diff
@@ -11,16 +11,42 @@
 
 # Following is template for the shell code injected into your session
 SHELLCODE = '''
-!cmd() {
+%cmd() {
     local d
-    d=$(!prog "$@")
+    # shellcheck disable=SC2164
+    d="$(%prog "$@")" && builtin cd -- "$d"
+}
 
-    if [ $? -ne 0 ]; then
-        return 0
-    fi
+# ======================================================================
+# Hook to add new entries to the history
 
-    builtin cd -- "$d"
-}
+if [[ -n $BASH_VERSION ]]; then
+    __cdhist_oldpwd="$(pwd)"
```

Comment:
does not make sense to save pwd when the cdhist is initialized in shell startup

Reply:
If you would not initialise it with the default cwd and you would execute a command that does not change the cwd, the condition `$__old_pwd != $pwd_tmp` would succed and the cwd would be added to the history although the cwd actually did not change.